packages/core/lib/services/dock.js

```diff
@@ -0,0 +1,46 @@
+import {DockAPI} from '@docknetwork/sdk';
+import {getCurrentPair} from './keyring';
+
+let isDockReady = false;
+
+export const dock = new DockAPI();
+
+export async function ensureDockReady() {
+  if (isDockReady) {
+    return;
+  }
+
+  return new Promise(resolve => {
+    const checkDockReady = () => {
+      if (isDockReady) {
+        return resolve();
+      }
+
+      setTimeout(checkDockReady, 200);
+    };
+
+    checkDockReady();
+  });
+}
+
+export default {
+  name: 'dock',
+  routes: {
+    async init(...params) {
+      const result = await dock.init(...params);
```

Comment:
any value in asserting inputs in this?

Reply:
I've added an assert for the address parameter. Probably we should have some better validation in the dock sdk as well.